std/string.d

```diff
@@ -368,22 +473,26 @@ ptrdiff_t indexOf(Char)(in Char[] s, in dchar c,
         {                                                   // Plain old ASCII
             auto c1 = cast(char) std.ascii.toLower(c);
 
-            foreach (ptrdiff_t i, c2; s)
+            ptrdiff_t i;
+            foreach (const c2; s.byCodeUnit())
             {
-                auto c3 = std.ascii.toLower(c2);
+                const c3 = std.ascii.toLower(c2);
                 if (c1 == c3)
```

Comment:
Could use `if (c1 == std.ascii.toLower(c2))` here to get rid of the awkward `c3` name, not important though.